src/client/api/object.c

```diff
@@ -162,6 +162,19 @@ daos_obj_query_key(daos_handle_t oh, daos_handle_t th, uint32_t flags,
 	return dc_task_schedule(task, true);
 }
 
+int
+daos_obj_sync(daos_handle_t oh, daos_epoch_t *epoch, daos_event_t *ev)
```

Comment:
I think I used wrong or confused term, it should not be caller "object barrier", instead, it should be called "object sync". On some degree, the semantics of "barrier" is not clear, sometimes it means that all processing will arrive at the barrier and wait there for others. But that is not what we need. Here, what we want is that force commit all the modifications that have ever been done before the specified epoch, and if some transactions have not ready to be committed, then restart them with higher epoch. So the semantics is more close to POSIX "sync" semantics. The "barrier" is just the side effect of "object sync".

> Another thing is that I don't think 'epoch/hlc' should be specified by user input for barrier, I'm even wondering if the barrier API should be exported to applications (I think it could be an DAOS internal API)?

Since POSIX "sync" is exported, then exporting DAOS "sync" seems also reasonable.